src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/ViewDataDictionaryControllerPropertyActivator.cs

```diff
@@ -81,9 +86,16 @@ private PropertyActivator<ControllerContext>[] GetPropertiesToActivate(Type type
 
         private ViewDataDictionary GetViewDataDictionary(ControllerContext context)
         {
-            return new ViewDataDictionary(
+            if (_viewDataDictionaryFactory != null)
+            {
```

Comment:
There is a null check in the ctor for `viewDataDictionaryFactory`. When is this condition not met?

Reply:
Cleaned this up. We'll initialize `_viewDataDictionaryFactory` in the obsolete controller so things keep working.